.github/workflows/codeql.yml

```diff
@@ -0,0 +1,147 @@
+        cd nettle-${nettleversion}
+        ./configure --enable-static --disable-shared --disable-openssl --disable-mini-gmp -disable-gcov --disable-documentation
+        sudo make -j $(nproc) install
+
+    - name: Install mbedTLS
+    # Build static mbedTLS with pthread support
+    # Disable AESNI on linux/386 asit would possibly result in an incompatible
+    # binary in processors lacking the AESNI and SSE2 instruction sets
+      run: |
+        curl -sSL https://ftl.pi-hole.net/libraries/mbedtls-${mbedtlsversion}.tar.bz2 | tar -xj
+        cd mbedtls-${mbedtlsversion}
+        sed -i '/#define MBEDTLS_THREADING_C/s*^//**g' include/mbedtls/mbedtls_config.h
+        sed -i '/#define MBEDTLS_THREADING_PTHREAD/s*^//**g' include/mbedtls/mbedtls_config.h
+        sudo make -j $(nproc) install
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v3
+      with:
+        languages: ${{ matrix.language }}
+        build-mode: ${{ matrix.build-mode }}
+        # If you wish to specify custom queries, you can do so here or in a config file.
+        # By default, queries listed here will override any specified in a config file.
+        # Prefix the list here with "+" to use these queries and those in the config file.
+
+        # For more details on CodeQL's query packs, refer to: https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning#using-queries-in-ql-packs
+        queries: security-and-quality
+
+    # If the analyze step fails for one of the languages you are analyzing with
+    # "We were unable to automatically build your code", modify the matrix above
+    # to set the build mode to "manual" for that language. Then modify this step
+    # to build your code.
+    # ℹ️ Command-line programs to run using the OS shell.
+    # 📚 See https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun
+    - if: matrix.build-mode == 'manual'
+      shell: bash
+      run: |
+        ./build.sh
+
+    - name: Perform CodeQL Analysis
+      uses: github/codeql-action/analyze@v3
+      with:
+        category: "/language:${{matrix.language}}"
+        upload: failure-only # upload only in case of failure, otherwise upload later after filtering
+        output: codeql-results
+    
+    - name: Filter SARIF
+      uses: advanced-security/filter-sarif@v1
+      with:
+        # filter out third-party dependencies
+        patterns: |
+          -src/dnsmasq/*
+          -src/webserver/civetweb/*
+          -src/webserver/cJSON/*
+          -src/tre-regex/*
+          -src/config/tomlc99/*
+          -src/database/shell.c
+          -src/database/sqlite3.c
+          -src/database/sqlite3.h
+          -src/zip/miniz/*
+          -src/lua/*
+          +src/lua/ftl_*  
```

Comment:
Do we need to exclude the files from `/src/api/docs/content/` _(there are no C files there)_ and include files from `/src/lua/scripts/`?

```suggestion
          +src/lua/ftl_*  
          +src/lua/scripts/*
          -src/api/docs/content/**/*
```

Reply:
I don't think we need to exclude the `api/docs`. The language to check is set to `c-cpp` so it should ignore all non C/C++ files. I also did not see any issues reported for `api/docs`.

Regarding `/src/lua/scripts/`: it looks like it's mainly copied from the original source with just a few lines added. But let's ask @DL6ER about inclusion yes/no.

Reply:
So far, it's indeed all vendor stuff